apps/framework-cli/src/infrastructure/redis/redis_client.rs

```diff
@@ -424,10 +407,30 @@ impl RedisClient {
     }
 
     pub async fn check_connection(&mut self) -> Result<()> {
-        redis::cmd("PING")
-            .query_async::<_, String>(&mut *self.connection.lock().await)
-            .await
-            .context("Failed to ping Redis server")?;
+        retry(
+            || async {
+                let result = std::panic::AssertUnwindSafe(async {
+                    redis::cmd("PING")
+                        .query_async::<_, String>(&mut *self.connection.lock().await)
+                        .await
+                })
+                .catch_unwind()
+                .await;
+
+                match result {
+                    Ok(Ok(_)) => Ok(()),
+                    Ok(Err(e)) => Err(e),
+                    Err(_) => Err(redis::RedisError::from(std::io::Error::new(
```

Comment:
Log this Err so that we have some insight into why it panicked?

Reply:
Agreed!